main.rb

```diff
@@ -0,0 +1,41 @@
+require 'io/console'
+
+class MainMenu
+  def initialize
+    ### Create the method named in the value of the hash to ask for
+    ### particular option parameters and instance external classes
+    @options = { '1' => 'list_all_books', '2' => 'list_music_albums',
+                 '3' => 'list_all_movies', '4' => 'list_of_games',
+                 '5' => 'list_all_genres', '6' => 'list_all_labels',
+                 '7' => 'list_all_authors', '8' => 'list_all_sources',
+                 'B' => 'add_book', 'A' => 'add_music_album',
+                 'M' => 'add_movie', 'G' => 'add_game', 'X' => 'exit_app' }
+  end
+
+  def show_options
+    # comment to test with RSPEC
+    $stdout.clear_screen
+    puts 'Welcome to Catalog of my things'
+    puts
+    puts 'Please choose an option by entering the number/letter:'
+    @options.each do |key, value|
+      puts "[#{key}] - #{value.gsub(/_/, ' ').capitalize!}"
+    end
+  end
+
+  def exit_app
+    puts 'Thank you for using the app!'
+    exit(0)
+  end
+
+  def main_menu
+    show_options
+    user_input = gets.chomp.to_s.upcase
+    exit_app if user_input == 'X'
+    send(@options[user_input])
```

Comment:
Please also add that when a user enters an invalid character, it should display an error message.

Reply:
ok well done